src/Unit-2/lesson3/README.md

```diff
@@ -0,0 +1,254 @@
+from the database. For this lesson, you just will need the names and ids.
+
+````csharp
+public class Employee
+{
+    public string Id { get; set; }
+    public string FirstName { get; set; }
+    public string LastName { get; set; }
+}
+
+public class ContactClass
+{
+    public string Name { get; set; }    
+}
+    
+public class Company
+{
+    public string Id { get; set; }
+    public ContactClass Contact { get; set;  }
+}
+
+public class Supplier
+{
+    public string Id { get; set; }
+    public ContactClass Contact { get; set; }
+}
+````
+
+### Step 3: Write the `Multi-map index definition`
+
+Everytime you create a multi-map index, the index definition class inherits from `AbstractMultiMapIndexCreationTask`.
+
+````csharp
+using System.Linq;
+using Raven.Abstractions.Indexing;
+using Raven.Client.Indexes;
+
+namespace MultimapIndexes
+{
+    public class People_Search : AbstractMultiMapIndexCreationTask<People_Search.Result>
+    {
+        public class Result
+        {
+            public string SourceId { get; set; }
+            public string Name { get; set; }
+            public string Type { get; set; }
+        }
+
+        public People_Search()
+        {
+            AddMap<Company>(companies =>
+                from company in companies
+                select new Result
+                {
+                    SourceId = company.Id,
+                    Name = company.Contact.Name,
+                    Type = "Company's contact"
+                }
+                );
+
+            AddMap<Supplier>(suppliers =>
+                from supplier in suppliers
+                select new Result
+                {
+                    SourceId = supplier.Id,
+                    Name = supplier.Contact.Name,
+                    Type = "Supplier's contact"
+                }
+                );
+
+            AddMap<Employee>(employees =>
+                from employee in employees
+                select new Result
+                {
+                    SourceId = employee.Id,
+                    Name = $"{employee.FirstName} {employee.LastName}",
+                    Type = "Employee"
+                }
+                );
+
+            Index(entry => entry.Name, FieldIndexing.Analyzed);
+
+            Store(entry => entry.SourceId, FieldStorage.Yes);
+            Store(entry => entry.Name, FieldStorage.Yes);
+            Store(entry => entry.Type, FieldStorage.Yes);
+        }
+    }
+}
+````
+
+You can define as many `map functions` as you need. Each map function is defined using
+the `AddMap` method and have to produce the same output type. The "source" collection is specified
+by the generic parameter type you specify in the `AddMap` function.   
+
+The `Index` method here was used to mark the `Name` class as `Analyzed`
+which enables full text search with this field.
+
+The `Store` method was used to enable projections. We will talk more about 
+that in this lesson.
```

Comment:
Actually the only sentence about projections that we can read later is "We will talk more about the "Projection" magic in the future."